tests/integration/linux_bridge_test.py

```diff
@@ -410,6 +425,120 @@ def test_port_vlan_not_implemented(port0_up):
+
+    def test_access_port_config(self, port0_up):
+        access_port_state = TestVlanFiltering._generate_vlan_filtering_config(
+            LinuxBridge.Port.Vlan.Mode.ACCESS, tag=TestVlanFiltering.access_tag
+        )
+        port_name = port0_up[Interface.KEY][0][Interface.NAME]
+        bridge_state = _create_bridge_subtree_config((port_name,))
+        bridge_state[LinuxBridge.PORT_SUBTREE][0].update(access_port_state)
+        with linux_bridge(TEST_BRIDGE0, bridge_state) as desired_state:
+            assertlib.assert_state(desired_state)
+
+    def test_access_port_update(self, port0_up, bridge_with_port_vlan_config):
+        port_name = port0_up[Interface.KEY][0][Interface.NAME]
+        bridge_state = _create_bridge_subtree_config((port_name,))
+
+        new_port_state = TestVlanFiltering._generate_vlan_filtering_config(
+            LinuxBridge.Port.Vlan.Mode.ACCESS,
+            tag=TestVlanFiltering.access_tag + 5,
+        )
+        bridge_state[LinuxBridge.PORT_SUBTREE][0].update(new_port_state)
+        with linux_bridge(TEST_BRIDGE0, bridge_state) as desired_state:
+            assertlib.assert_state(desired_state)
+
+    def test_trunk_ports_update(self, port0_up, bridge_with_port_vlan_config):
+        port_name = port0_up[Interface.KEY][0][Interface.NAME]
+        bridge_state = _create_bridge_subtree_config((port_name,))
+
+        new_trunk_tags = TestVlanFiltering._generate_vlan_id_config(600, 602)
+        new_port_state = TestVlanFiltering._generate_vlan_filtering_config(
+            LinuxBridge.Port.Vlan.Mode.TRUNK, trunk_tags=new_trunk_tags
+        )
+        bridge_state[LinuxBridge.PORT_SUBTREE][0].update(new_port_state)
+        with linux_bridge(TEST_BRIDGE0, bridge_state) as desired_state:
+            assertlib.assert_state_match(desired_state)
+
+    @staticmethod
+    def _generate_vlan_filtering_config(
+        port_type, trunk_tags=None, tag=None, native_vlan=None
+    ):
+        vlan_filtering_state = {
+            LinuxBridge.Port.Vlan.MODE: port_type,
+            LinuxBridge.Port.Vlan.TRUNK_TAGS: trunk_tags or [],
+        }
+
+        if tag:
+            vlan_filtering_state[LinuxBridge.Port.Vlan.TAG] = tag
+        if native_vlan is not None:
+            vlan_filtering_state[
+                LinuxBridge.Port.Vlan.ENABLE_NATIVE
+            ] = native_vlan
+
+        return {LinuxBridge.Port.VLAN_SUBTREE: vlan_filtering_state}
+
+    @staticmethod
+    def _generate_vlan_id_config(*vlan_ids):
+        return [
+            {LinuxBridge.Port.Vlan.TrunkTags.ID: vlan_id}
+            for vlan_id in vlan_ids
+        ]
+
+    @staticmethod
+    def _generate_vlan_id_range_config(min_vlan_id, max_vlan_id):
+        return {
+            LinuxBridge.Port.Vlan.TrunkTags.ID_RANGE: {
+                LinuxBridge.Port.Vlan.TrunkTags.MIN_RANGE: min_vlan_id,
+                LinuxBridge.Port.Vlan.TrunkTags.MAX_RANGE: max_vlan_id,
+            }
+        }
+
+
```

Comment:
IMHO, this should be move to `tests/integration/testlib/bridgelib.py` as you did in unit tests.

Reply:
I don't think these functions will be re-used in other integ tests.

If I'm wrong, they can be moved when the need arises. Are you OK with that ?

Reply:
I think it needs to be well integrated into the existing helpers (e.g. `add_port_to_bridge`).
But we can do this later on, it does not worth blocking the PR for it.